daemon/src/project/barebone.rs

```diff
@@ -53,29 +53,33 @@ impl ProjectRun for BareboneProject {}
 impl ProjectCompile for BareboneProject {
     async fn update_compile_database(&self) -> Result<()> {
         use xclog::XCCompilationDatabase as CC;
+        use xclog::XCCompileCommand as C;
 
         let (name, root) = (self.name(), self.root());
         let cache_root = self.build_cache_root()?;
+        let mut compile_commands: Vec<C> = vec![];
+
         let mut args = self.compile_arguments();
 
         args.push(format!("SYMROOT={cache_root}"));
 
         let xcworkspace = format!("{name}.xcworkspace");
 
         if self.root().join(&xcworkspace).exists() {
-            args.extend_from_slice(&[
-                "-workspace".into(),
-                xcworkspace,
-                "-scheme".into(),
-                name.into(),
-            ]);
+            args.extend_from_slice(&["-workspace".into(), xcworkspace]);
+
+            for scheme in get_workspace_schemes(self.root(), &args).await?.into_iter() {
+                let mut args = args.clone();
+                args.extend_from_slice(&["-scheme".into(), scheme]);
+                log::info!("xcodebuild {}", args.join(" "));
+                compile_commands.extend(CC::generate(&root, &args).await?.to_vec());
```

Comment:
Is this going to run xcodebuild for every scheme? Or what is this doing?

Reply:
yes

Reply:
It takes longer too, I'm think of skipping anything Pods

Reply:
> yes

Okay so, I don't think this is the right solution. What the user wants is to build a single target using a single scheme, not to try out building with each scheme just to see/go with what works. Instead, we should do something along the line of what Xcode does:

![sfdafdafa](https://user-images.githubusercontent.com/46855713/175374201-db14a164-1d0e-43e6-9f3e-d73ce53b9aef.png)

You can see here that Xcode lets you choose the current scheme, but by default I think the one you want is the same as the target name. So what I think we should do, is just by default assume the target name == the scheme name and use that, but also create a new picker that lets the user select which scheme to use just in case that is a poor assumption.

I think this change should be tackled in another PR though, rather than grouped into this one. I'd be happy to try my hand at doing this myself, but it'll have to be a bit before I could, because I don't have time to write the code right now.

Reply:
I don't think this issue really blocks this PR though, with my non-multiplatform project it seemed to work fine. Just would want to revert https://github.com/xbase-lab/xbase/pull/101/commits/2b303459a62bf1a79ab25d53f260e7488f6bf51d before merging.

Reply:
> I think this change should be tackled in another PR though, rather than grouped into this one. I'd be happy to try my hand at doing this myself, but it'll have to be a bit before I could, because I don't have time to write the code right now.

Oh, alright @smolck, [diff](https://paste.sh/EOpwjihV#mlB5a5CXihPI9FvZ7K0WSQNn) I'm going to work on removing neovim logic from daemon instead. Good luck 🌝

Reply:
use this diff instead https://paste.sh/HDtzcK3r#98AMlOsPX2cZotAdEh0_wNHb , the above is incorrect